recommender/static/js/src/cats.js

```diff
@@ -1,187 +1,203 @@
-    '.recommender_resourceRankingForRemovalButton.recommender_removeMode': '<span>Click to view resources in ordinary decreasing-vote order</span>'
-};
-
-var problematicReasonsPrefix = '<br/>Here is a list of reasons why students think this resource problematic: <br/>&nbsp;&nbsp;&nbsp;&nbsp;';
-var endorsedReasonsPrefix = '<br/>The reason why it is endorsed is: <br/>&nbsp;&nbsp;&nbsp;&nbsp;';
-var recommenderResourceAriaPrefix = 'Resource: ';
-var recommenderVoteScorePostfix = ' votes';
-var reasonSeparator = '<br/>&nbsp;&nbsp;&nbsp;&nbsp;';
-
-var exportResourceFileInfo = {
-    'fileType': 'data:application/json;charset=utf-8,',
-    'fileName': 'resource.json'
-};
-
-var confirmInterruptSubmission = 'The content you typed has not been submitted yet. Are you sure to go back?';
-
-var headerText = {
-    '.recommender_importResourcePage': 'Import resources',
-    '.recommender_addResourcePage': 'Suggest resource',
-    '.recommender_editResourcePage': 'Edit existing resource',
-    '.recommender_flagResourcePage': 'Flag Resource',
-    '.recommender_endorsePage': 'Endorse Resource',
-    '.recommender_removePage': 'Remove Resource'
-};
-
-var modifyPageTitle = {
-    '.recommender_importResourcePage': 'Upload resources in JSON format to the database.',
-    '.recommender_addResourcePage': 'Suggest a resource which can help other students with this problem. Please do not give the answer directly.',
-    '.recommender_editResourcePage': 'Edit the resource and make it more helpful for other students with this problem. Please do not give the answer directly.',
-    '.recommender_flagResourcePage': 'Why would you like to flag this resource? The staff will review all flagged resources, and remove inappropriate ones (spam, incorrect, abusive, etc.). Giving a clear reason will help us do this efficiently.',
-    '.recommender_endorsePage': 'Endorse this resource and give the reason why you do that.',
-    '.recommender_removePage': 'Remove this resource and give the reason why you do that.'
-};
-
-var writeDatabaseEnum = {
-    ADD: 'add',
-    EDIT: 'edit'
-};
-
-var voteTypeEnum = {
-    UPVOTE: 'upvote',
-    DOWNVOTE: 'downvote'
-};
-
-var sortResourceEnum = {
-    INCREASE: 'increase',
-    DECREASE: 'decrease'
-};
-
-var voteConfigs = {
-    'upvote': {
-        'buttonClassName': 'recommenderVoteArrowUp',
-        'eventName': 'arrowUp',
-        'serverEventName': 'recommender_upvote',
-        'voteClassName': 'recommender_upvoting',
-        'previousVoteClassName': 'recommender_downvoting'
-    },
-    'downvote': {
-        'buttonClassName': 'recommenderVoteArrowDown',
-        'eventName': 'arrowDown',
-        'serverEventName': 'recommender_downvote',
-        'voteClassName': 'recommender_downvoting',
-        'previousVoteClassName': 'recommender_upvoting'
+(function (window) {
+    var gettext = window.gettext || (function (string) {
```

Comment:
```suggestion
   // Wrapping this file in an Immediately Invoked Function (IIAF) to ensure that helpers such as `gettext` and `span` aren't poluting the global scope.
   //
   // Note: The global `window.*` variables should be converted into local ones to avoid over-using global variables. This is a 2014-era legacy XBlock coding standard that should be refactored -- @OmarIthawi at Apr 4, 2023 
   //
    var gettext = window.gettext || (function (string) {
```